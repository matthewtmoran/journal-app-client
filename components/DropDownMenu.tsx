import React from "react";
import {
  Text,
  View,
  StyleSheet,
  TouchableWithoutFeedback,
  TouchableOpacity,
} from "react-native";
import Modal from "react-native-modal";
import { RobotText } from "./StyledText";

interface IItem {
  label: string;
  action(): void;
}

interface IProps {
  isVisible: boolean;
  items: IItem[];
  hide(): void;
}

const DropDownMenu = ({ items, isVisible, hide }: IProps) => {
  return (
    <Modal
      animationIn="slideInDown"
      animationOut="slideOutUp"
      hasBackdrop={false}
      isVisible={isVisible}
    >
      <TouchableWithoutFeedback onPress={() => hide()}>
        <View style={styles.container}>
          <View style={[styles.menu, { top: 0, right: 0 }]}>
            {items.map(({ label, action }) => {
              return (
                <TouchableOpacity
                  key={label}
                  style={{ ...styles.textContainer }}
                  onPress={action}
                >
                  <RobotText style={styles.text}>{label}</RobotText>
                </TouchableOpacity>
              );
            })}
          </View>
        </View>
      </TouchableWithoutFeedback>
    </Modal>
  );
};

export default DropDownMenu;

const styles = StyleSheet.create({
  container: {
    position: "absolute",
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    zIndex: 9999,
  },
  menu: {
    position: "absolute",
    backgroundColor: "white",
    textAlign: "left",
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 5,
    },
    shadowOpacity: 0.36,
    shadowRadius: 6.68,
    elevation: 11,
    width: "50%",
    zIndex: 9999,
    display: "flex",
    alignItems: "flex-start",
  },
  textContainer: {
    flex: 1,
    textAlign: "left",
    alignItems: "flex-start",
    paddingHorizontal: 14,
    paddingVertical: 8,
    width: "100%",
  },
  text: {
    textAlign: "left",
    fontSize: 16,
  },
});
