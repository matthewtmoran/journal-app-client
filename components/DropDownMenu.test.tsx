import React from "react";
import { render } from "@testing-library/react-native";
import DropDownMenu from "./DropDownMenu";

describe("components/DropDownMenu", () => {
  const props = {
    isVisible: false,
    hide: jest.fn(),
    items: [
      { label: "Edit", action: jest.fn() },
      {
        label: "Remove",
        action: jest.fn(),
      },
    ],
  };

  it("Should match snapshot", () => {
    const { baseElement } = render(<DropDownMenu {...props}></DropDownMenu>);
    expect(baseElement).toMatchSnapshot();
  });

  it("Should render items", async () => {
    const { findByText } = render(<DropDownMenu {...props}></DropDownMenu>);
    expect(findByText("Edit")).toBeTruthy();
    expect(findByText("Remove")).toBeTruthy();
  });
});
