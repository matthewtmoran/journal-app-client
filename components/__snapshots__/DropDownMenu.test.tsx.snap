// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/DropDownMenu Should match snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      hideModalContentWhileAnimating={false}
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
      transparent={true}
      visible={false}
    >
      <View
        hideModalContentWhileAnimating={false}
        pointerEvents="box-none"
        scrollHorizontal={false}
        scrollOffset={0}
        scrollOffsetMax={0}
        scrollTo={null}
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "margin": 37.5,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        swipeThreshold={100}
      >
        <TouchableWithoutFeedback>
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 9999,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-start",
                    "backgroundColor": "white",
                    "display": "flex",
                    "elevation": 11,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 5,
                      "width": 0,
                    },
                    "shadowOpacity": 0.36,
                    "shadowRadius": 6.68,
                    "textAlign": "left",
                    "width": "50%",
                    "zIndex": 9999,
                  },
                  Object {
                    "right": 0,
                    "top": 0,
                  },
                ]
              }
            >
              <TouchableOpacity
                activeOpacity={0.2}
                style={
                  Object {
                    "alignItems": "flex-start",
                    "flex": 1,
                    "paddingHorizontal": 14,
                    "paddingVertical": 8,
                    "textAlign": "left",
                    "width": "100%",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 16,
                        "textAlign": "left",
                      },
                      Object {
                        "fontFamily": "roboto",
                      },
                    ]
                  }
                >
                  Edit
                </Text>
              </TouchableOpacity>
              <TouchableOpacity
                activeOpacity={0.2}
                style={
                  Object {
                    "alignItems": "flex-start",
                    "flex": 1,
                    "paddingHorizontal": 14,
                    "paddingVertical": 8,
                    "textAlign": "left",
                    "width": "100%",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 16,
                        "textAlign": "left",
                      },
                      Object {
                        "fontFamily": "roboto",
                      },
                    ]
                  }
                >
                  Remove
                </Text>
              </TouchableOpacity>
            </View>
          </View>
        </TouchableWithoutFeedback>
      </View>
    </Modal>
  </View>
</View>
`;
